File: A3/A3-015/A3-015.py
# Input pillar amount
N = int(input())
# Input pillar length
pillars = []
for _ in range(N):
    pillars.append(int(input()))

# Sort pillars length in ascending order
pillars.sort()
# Calculate total distance
curr = 0
dist = 0
for L in pillars:
    curr += L           # Calculate current total pillar length
    dist += curr * 2    # Calculate total distance
# Output
print(dist)

File: A3/A3-015/A3-015.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int N, dist = 0;
vector<int> pillars;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    // Input pillar amount
    cin >> N;
    pillars.resize(N);
    // Input pillar length
    for(int i = 0; i < N; i++) { cin >> pillars[i]; }
    // Sort pillar length in ascending order
    sort(pillars.begin(), pillars.end());
    // Calculate total distance
    int curr = 0;
    for(int i = 0; i < N; i++)
    {
        curr += pillars[i]; // Calculate current total pillar length
        dist += curr * 2;   // Calculate total distance
    }
    // Output
    cout << dist << '\n';
    return 0;
}